### ISSUE TITLE
Plugin totally messes up on new posts now.

### ISSUE THREAD

**Issue by metallikat36** [dsifford/academic-bloggers-toolkit#373]
See the video:
https://youtu.be/iRcyxsuRIN4
Not sure if this started with the last update or not. Weird. 

The references are overwriting themselves, appearing on the uncited list, and doing other weird things. Also got an error: "Cannot read property "id" of undefined".

Reminder, you have my site login, if you want a place to reproduce. 

No rush. Just happy you are always so willing to develop this plugin.

**Comment by dsifford** [comment]
Sorry to hear that!

If you are able to reliably reproduce the issue, please write down the exact steps you take to trigger the error and I'll look into it.

**Comment by metallikat36** [comment]
The steps are exactly what is in the video. Just make a new page or post and try to do anything at all with the plugin. It won't work adding beyond 1 reference.

**Comment by dsifford** [comment]
I'm able to reproduce. Gonna work on simplifying how citations are checked for duplicates by using non-cryptographic hashes rather than complex logic. I think that might help.

Also working on fixing issues with IE11. Next update should be out by next weekend if time is on my side.

**Comment by dsifford** [comment]
@metallikat36 I think I got it sorted out. Spent most of the day working on perf improvements. Would you be interested in testing it out if I compile a copy for you prior to pushing the update?

**Comment by metallikat36** [comment]
Sure, I can test it. I just deactivate and uninstall the current version, then upload the new version, right?

**Comment by dsifford** [comment]
Correct. (Keeping in mind that you'll lose any options you set in the options page if you physically uninstall the plugin... Deleting it manually won't trigger uninstall hooks).

You might need to make sure your cache is cleared since I didn't change the version number in this build.

Thanks for the help.

[academic-bloggers-toolkit-prerelease.zip](https://github.com/dsifford/academic-bloggers-toolkit/files/1405551/academic-bloggers-toolkit-prerelease.zip)

**Comment by metallikat36** [comment]
Okay, I will test this tomorrow.

**Comment by metallikat36** [comment]
Doesn't work. The Reference list has no options when I make a new post:
![image](https://user-images.githubusercontent.com/17968340/31921644-4ee5470e-b825-11e7-9628-bcf0117830ca.png)

**Comment by dsifford** [comment]
@metallikat36 That's likely cache. I didn't change the version number so you need to make sure you manually clear your server cache.

Did you do that?

**Comment by metallikat36** [comment]
Clearing just my server cache wasn't enough. I cleared by browser cache and cookies, and disabled CloudFlare, and then the plugin options showed up as normal.

But, still buggy. The following video is a little long (4 minutes), but it gives step-by-step things you can try to reproduce, with PMID numbers. Somethings worked, some things didn't. This will need another round of testing for sure. 
https://youtu.be/AVj6XAV_pKo

**Comment by dsifford** [comment]
Thanks for the vid. Very helpful.

Most of those things are broken downstream in TinyMCE. I'll see if I can maybe improve on my end, but more than likely it'll need to be fixed by the tinymce people.

The PMC is definitely ABT domain though so I'll get that sorted out.

**Comment by metallikat36** [comment]
Maybe I haven't been paying enough attention, but is the tinymce stuff you are referring to something I should expect to be fixed in a future update of wordpress or something? Or is there someone that I need to be contacting?

**Comment by dsifford** [comment]
Not worth the effort. WordPress is currently at version 4.7. When WordPress hits version 5.0 (expected early in 2018) they are switching from TinyMCE to their [Gutenberg editor](https://github.com/WordPress/gutenberg/). And notably, when that happens, this plugin will be bricked completely until I can see if I can figure out how to make it work.

I've prepared as much as I can for the Gutenberg switch, but their API is still not stable enough to warrant working any further. It's clear though already that it'll be a much more complex endeavor because instead of one single editor to manage, each "block" that the user creates will have to be managed independently. 

TL;DR: Not worth fixing the tinymce issues. Gutenburg is right around the corner.

**Comment by dsifford** [comment]
Looking at this now...

To comment directly in order of the issues you list...

1. TinyMCE issue, confirmed.
2. TinyMCE issue, confirmed.
3. ABT issue. I'll fix.
3. "Around 2:30 tried selecting a block" -- It should still work even though the block doesn't appear surrounded. (Again, TinyMCE issue)

In summary: The only fixable issue in that video is the PMC thing. All others are annoyances that we're going to have to live with for now because of the imperative nature of TinyMCE and the associated weaknesses that come with imperative programming.

**Comment by dsifford** [comment]
@metallikat36 Might be worth bringing up issue 1 with tinymce now that I look at it. That's sort of a big deal.

I'll see if I can chat with them about it when I get time. The issue is that their "selection" api is broken and doesn't produce the correct `Range`.
